src/app/search/SearchBar.tsx

```diff
@@ -0,0 +1,362 @@
+);
+
+// TODO: break filter autocomplete into its own object/helpers... with tests
+
+/** An interface for interacting with the search filter through a ref */
+export interface SearchFilterRef {
+  /** Switch focus to the filter field */
+  focusFilterInput(): void;
+  /** Clear the filter field */
+  clearFilter(): void;
+}
+
+/**
+ * A reusable, autocompleting item search input. This is an uncontrolled input that
+ * announces its query has changed only after some delay. This is the new version of the component
+ * that offers a browser-style autocompleting search bar with history.
+ *
+ * TODO: Should this be the main search bar only, or should it also work for item picker, etc?
+ */
+export default React.forwardRef(function SearchFilterInput(
+  {
+    searchQueryVersion,
+    searchQuery,
+    placeholder,
+    children,
+    autoFocus,
+    onQueryChanged,
+    onClear,
+  }: Props,
+  ref: React.Ref<SearchFilterRef>
+) {
+  const [liveQuery, setLiveQuery] = useState('');
+  const [filterHelpOpen, setFilterHelpOpen] = useState(false);
+  const dispatch = useDispatch();
+  const autocompleter = useSelector((state: RootState) =>
+    createAutocompleter(searchConfigSelector(state))
+  );
+  const isPhonePortrait = useSelector(isPhonePortraitSelector);
+  const recentSearches = useSelector(recentSearchesSelector);
+  const inputElement = useRef<HTMLInputElement>(null);
+  const [items, setItems] = useState(() => autocompleter(liveQuery, 0, recentSearches));
+  // TODO: this isn't great. We need https://github.com/downshift-js/downshift/issues/1144
+  useEffect(() => {
+    setItems(autocompleter(liveQuery, inputElement.current?.selectionStart || 0, recentSearches));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [recentSearches]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedUpdateQuery = useCallback(
+    _.debounce((query: string) => {
+      onQueryChanged(query);
+    }, 500),
+    [onQueryChanged]
+  );
+
+  const focusFilterInput = useCallback(() => {
+    inputElement.current?.focus();
+  }, []);
+
+  const onBlur = () => {
+    if (liveQuery) {
+      // save this to the recent searches only on blur
+      dispatch(searchUsed(liveQuery));
+    }
+  };
+
+  // Is the current search saved?
+  const canonical = canonicalizeQuery(parseQuery(liveQuery));
+  const saved = recentSearches.find((s) => s.query === canonical)?.saved;
+
+  const toggleSaved = () => {
+    // TODO: keep track of the last search, if you search for something more narrow immediately after then replace?
+    dispatch(saveSearch({ query: liveQuery, saved: !saved }));
+  };
+
+  // useCombobox from Downshift manages the state of the dropdown
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getMenuProps,
+    getInputProps,
+    getLabelProps,
+    getComboboxProps,
+    highlightedIndex,
+    getItemProps,
+    reset,
+  } = useCombobox<SearchItem>({
+    items,
+    isOpen: true,
```

Comment:
```suggestion
```

Reply:
Whoops! I was forcing it open for testing